### PR TITLE
Add export-history subcommand.

### PR DESCRIPTION
**Pull request by henridf**
`geth export-history` exports blocks and receipts in SSZ format.

running todo:
- [x] Add logging 
- [x] Check cli params: start < end 
- [x] Print out hash root for info purposes (it can also be computed with [`bart`](https://github.com/henridf/eip44s-proto#bart-cli-tool) )

### CODE REVIEW

**Comment by lightclient** [comment]
I rebased to `master`.

File: cmd/utils/cmd.go [hunk 1]
```diff
@@ -262,6 +264,96 @@ func ExportChain(blockchain *core.BlockChain, fn string) error {
...
+	name = name + fmt.Sprintf("-%d", n) + suffix
+	return name
+}
+
+func ExportHistory(bc *core.BlockChain, fn string, targetSize int) error {
+	return ExportHistoryRange(bc, fn, uint64(0), bc.CurrentBlock().NumberU64(), targetSize)
+}
+
+func ExportHistoryRange(bc *core.BlockChain, fn string, first uint64, last uint64, targetSize int) error {
+	var blocks []*spec.Block
```
[the remark this answers is not in the document]
Comment (lightclient) [reply cmd/utils/cmd.go:279]: Hmm is there a reason to define another block type if we annotate `types.Block` here?

File: cmd/utils/cmd.go [hunk 2]
```diff
@@ -262,6 +264,96 @@ func ExportChain(blockchain *core.BlockChain, fn string) error {
...
+	name := strings.TrimSuffix(basename, suffix)
+	name = name + fmt.Sprintf("-%d", n) + suffix
+	return name
+}
+
+func ExportHistory(bc *core.BlockChain, fn string, targetSize int) error {
+	return ExportHistoryRange(bc, fn, uint64(0), bc.CurrentBlock().NumberU64(), targetSize)
+}
+
+func ExportHistoryRange(bc *core.BlockChain, fn string, first uint64, last uint64, targetSize int) error {
```
Comment (lightclient) [inline cmd/utils/cmd.go:278]: I think there should be verification that `first+last < maxBlocks`?

**Review by lightclient — CHANGES_REQUESTED** [review]

File: cmd/utils/cmd.go (hunk 2 above)
Comment (henridf) [reply cmd/utils/cmd.go:278]: 🤔 what would maxBlocks here here?

File: cmd/utils/cmd.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (henridf) [reply cmd/utils/cmd.go:279]: > We should define the block here to target the spec and not define externally.

Happy to move it here (maybe in its own package under cmd/utils) If you're ok with adding that here (along with all the go.mod dependency additions)

> Hmm is there a reason to define another block type if we annotate types.Block here?

Back when I was defining that it seemed hard to do it all via just annotations on the existing geth structs, but I've paged out all the details 😬. Will take another look (maybe it is possible?) and report back.

**Comment by henridf** [comment]
@lightclient I moved the ssz definitions into the repo. 

Unfortunately right now annotating `types.Block` isn't possible due to some current limitations of fastssz:
- can't handle the `common.Hash` and `common.Address` types in `Header`  (https://github.com/ferranbt/fastssz/issues/108 - I will take a stab at at).
- big.Ints aren't supported (https://github.com/ferranbt/fastssz/issues/107)

**Comment by holiman** [comment]
Superseded by #26621

**Pull request closed without merging**